config/stored_requests.go

```diff
@@ -11,44 +10,31 @@ import (
 	"github.com/golang/glog"
 )
 
-// StoredRequests configures the backend used to store requests on the server.
-type StoredRequests struct {
-	// Files should be true if Stored Requests should be loaded from the filesystem.
-	Files bool `mapstructure:"filesystem"`
-	//If data should be loaded from file system, path should be specified in configuration
-	Path string `mapstructure:"directorypath"`
-	// Postgres configures Fetchers and EventProducers which read from a Postgres DB.
-	// Fetchers are in stored_requests/backends/db_fetcher/postgres.go
-	// EventProducers are in stored_requests/events/postgres
-	Postgres PostgresConfig `mapstructure:"postgres"`
-	// HTTP configures an instance of stored_requests/backends/http/http_fetcher.go.
-	// If non-nil, Stored Requests will be fetched from the endpoint described there.
-	HTTP HTTPFetcherConfig `mapstructure:"http"`
-	// InMemoryCache configures an instance of stored_requests/caches/memory/cache.go.
-	// If non-nil, Stored Requests will be saved in an in-memory cache.
-	InMemoryCache InMemoryCache `mapstructure:"in_memory_cache"`
-	// CacheEventsAPI configures an instance of stored_requests/events/api/api.go.
-	// If non-nil, Stored Request Caches can be updated or invalidated through API endpoints.
-	// This is intended to be a useful development tool and not recommended for a production environment.
-	// It should not be exposed to public networks without authentication.
-	CacheEventsAPI bool `mapstructure:"cache_events_api"`
-	// HTTPEvents configures an instance of stored_requests/events/http/http.go.
-	// If non-nil, the server will use those endpoints to populate and update the cache.
-	HTTPEvents HTTPEventsConfig `mapstructure:"http_events"`
-}
+// DataType constants
+type DataType string
 
-// StoredRequestsSlim struct defines options for stored requests from a single endpoint
-type StoredRequestsSlim struct {
+const (
+	RequestDataType    DataType = "Request"
+	CategoryDataType   DataType = "Category"
+	VideoDataType      DataType = "Video"
+	AMPRequestDataType DataType = "Amp Request"
+)
+
+// StoredRequests struct defines options for stored requests for each data type
+// including some amp stored_requests options
+type StoredRequests struct {
+	// DataType is a tag pushed from upstream indicating the type of object fetched here
+	DataType DataType
```

Comment:
We talked about this a bit offline, including making the DataType private and using an accessor - which is better, but I feel this is still mixing the concerns of config and context. Please consider a refactor where the host config is separated from the mix of host and hardcoded values.

Reply:
I have been thinking about how to do this in a concise way. The intention is that any method looking at this config can tell what it is about, and the current code is pretty clear about that. Wrapping a struct around it can also achieve this but adds complexity to any sender which must do the wrapping.

For now, I made `dataType` private with accessor (commit a603974), as a compromise. I've updated #1426 to use the same approach (https://github.com/prebid/prebid-server/pull/1426/commits/dc8a74d1bab82b87095832e78e284dadc8e88932) and probably the blacklist maps can eventually use this too.